File: string-paradigm-dependency_2/src/main/java/com/jesperancinha/string/paradigm/performance2/ParadigmDependency2TableModel.java
package com.jesperancinha.string.paradigm.performance2;

import java.util.ArrayList;
import java.util.List;

import javax.swing.table.AbstractTableModel;


public class ParadigmDependency2TableModel extends AbstractTableModel {

    /**
	 * 
	 */
	private static final long serialVersionUID = -7320366591500286562L;
	private List<ParadigmDependency2Impl> dependencies = new ArrayList<ParadigmDependency2Impl>();
    private int columnCount;
    
    public int addCombination(ParadigmDependency2Impl dependency) {
        this.dependencies.add(dependency);
        
        int codeCount = dependency.getDependencyCount();
        if(codeCount != this.columnCount) {
            this.columnCount = Math.max(columnCount, codeCount);
//            fireTableStructureChanged();
        }
        
        int row = dependencies.size() - 1;
//        fireTableRowsInserted(row, row);
        return row;
    }

    @Override
    public int getRowCount() {
        return dependencies.size();
    }

    @Override
    public int getColumnCount() {
        return columnCount;
    }

    @Override
    public Object getValueAt(int rowIndex, int columnIndex) {
        ParadigmDependency2Impl dependency = dependencies.get(rowIndex);
        return dependency.getSubDependencyElement(columnIndex).intern();
    }

    public void clear() {
        int size = this.dependencies.size();
        if(size > 0) {
            this.dependencies.clear();
            fireTableRowsDeleted(0, size - 1);
        }
    }
}

File: string-paradigm-dependency_2/src/main/kotlin/org/jesperancinha/string/paradigm/performance2/ParadigmDependency2TableModel.kt
package org.jesperancinha.string.paradigm.performance2

import javax.swing.table.AbstractTableModel

class ParadigmDependency2TableModel : AbstractTableModel() {
    private val dependencies: MutableList<ParadigmDependency2Impl> = ArrayList()
    private var columnCount = 0
    fun addCombination(dependency: ParadigmDependency2Impl): Int {
        dependencies.add(dependency)
        val codeCount = dependency.dependencyCount()
        if (codeCount != columnCount) {
            columnCount = columnCount.coerceAtLeast(codeCount)
            //            fireTableStructureChanged();
        }
        //        fireTableRowsInserted(row, row);
        return dependencies.size - 1
    }

    override fun getRowCount() = dependencies.size

    override fun getColumnCount() = columnCount

    override fun getValueAt(rowIndex: Int, columnIndex: Int) =
        dependencies[rowIndex].getSubDependencyElement(columnIndex).intern()

    fun clear() {
        val size = dependencies.size
        columnCount = 0
        if (size > 0) {
            dependencies.clear()
            fireTableRowsDeleted(0, size - 1)
        }
    }
}